statsmodels/stats/tests/test_multicollinearity.py

```diff
@@ -0,0 +1,293 @@
+
+        mcoll = MultiCollinearitySequential(self.x)
+
+        assert_allclose(mcoll.partial_corr, rsquared0, rtol=1e-12, atol=1e-15)
+        # infs could be just large values because of floating point imprecision
+        #assert_allclose(mcoll.vif, vif0, rtol=1e-13)
+        #mask_inf = np.isinf(vif0) & ~np.isinf(mcoll.vif)
+        #assert_allclose(mcoll.vif[~mask_inf], vif0[~mask_inf], rtol=1e-13)
+        #assert_array_less(1e30, mcoll.vif[mask_inf])
+        assert_allclose_large(mcoll.vif, vif0, rtol=1e-13, ltol=1e-14)
+
+        if not np.isinf(vif0).any():
+            # The following requires nonsingular matrix because of Cholesky
+            # check moment matrix as input
+            x_dm = x - x.mean(0)  # standardize doesn't demean
+            mcoll2 = MultiCollinearitySequential(None,
+                                                 moment_matrix=x_dm.T.dot(x_dm))
+            assert_allclose(mcoll2.partial_corr, mcoll.partial_corr, rtol=1e-13)
+            assert_allclose(mcoll2.vif, mcoll.vif, rtol=1e-13)
+
+            # check correlation matrix as input
+            mcoll2 = MultiCollinearitySequential(None,
+                                                 np.corrcoef(x, rowvar=False),
+                                                 standardize=False)
+            assert_allclose(mcoll2.partial_corr, mcoll.partial_corr, rtol=1e-13)
+            assert_allclose(mcoll2.vif, mcoll.vif, rtol=1e-13)
+
+        # Note we need a constant since x is not demeaned
+        collinear_columns, keep_columns = collinear_index(xf)
+        not_coll = [i for i in range(xf.shape[1]) if not i in collinear_columns]
+        assert_equal(not_coll, keep_columns)
+        # I haven't checked what the equvalent threshold is exactly
+        # subtracting 1 from index to ignore constant column
+        assert_equal(collinear_columns - 1, np.nonzero(mcoll.vif > 1e14)[0])
+
+
+    def test_multicoll(self):
+        xf = np.asarray(self.xf)  # convert from pandas DataFrame, for OLS only
+        nobs, k_vars = self.xf.shape
+        ols_results = []
+
+        for k in range(1, k_vars):
+            idx_k = list(range(k_vars))
+            del idx_k[k]
+            ols_results.append(OLS(xf[:,k], xf[:, idx_k]).fit())
+
+        rsquared0 = np.array([res.rsquared for res in ols_results])
+        vif0 = 1. / (1. - rsquared0)
+        # for checking singular cases with pdb in pytest
+        # if ((1. - rsquared0)<1e-14).any():
+        #     raise
+
+        mcoll = MultiCollinearity(self.x)
+
+        assert_allclose(mcoll.partial_corr, rsquared0, rtol=1e-13)
+        assert_allclose_large(mcoll.vif, vif0, rtol=1e-12, ltol=1e12)
+
+        vif1_ = vif(self.x)
+        vif1 = np.asarray(vif1_)   # check values if pandas.Series
+        # TODO: why does mcoll.vif have infs but vif1 doesn't?
+        assert_allclose_large(vif1, mcoll.vif, rtol=1e-12, ltol=1e12)
+        assert_allclose_large(vif1, vif0, rtol=1e-12, ltol=1e12)
```

Comment:
there might be a problem if we have inf or very large where we shouldn't
e.g. if all vif where set to inf, then the assert is empty